django/db/models/sql/compiler.py

```diff
@@ -150,7 +147,14 @@ def collapse_group_by(self, expressions, having):
                         getattr(expr.output_field, 'model') == self.query.model):
                     pk = expr
             if pk:
+                # MySQLism: Columns in HAVING clause must be added to the GROUP BY.
```

Comment:
The reason its here is that it has been implemented more or less this way for a long time (though the code hasn't been even this clear before compiler expressions refactor).

MySQL has been the only backend to reach this part of code. If some backend would allow grouping by primary key only, and wouldn't need expressions in having either, then that backend would get extra expressions in GROUP BY. This is non-optimal, but not an error.

Of course, if somebody wants to clean this up, +1 to that.